x-pack/plugins/beats/server/routes/api/register_verify_beats_route.js

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Joi from 'joi';
+import moment from 'moment';
+import {
+  get,
+  flatten
+} from 'lodash';
+import { INDEX_NAMES } from '../../../common/constants';
+import { callWithRequestFactory } from '../../lib/client';
+import { wrapEsError } from '../../lib/error_wrappers';
+
+async function getBeats(callWithRequest, beatIds) {
+  const ids = beatIds.map(beatId => `beat:${beatId}`);
+  const params = {
+    index: INDEX_NAMES.BEATS,
+    type: '_doc',
+    body: { ids },
+    _sourceInclude: [ 'beat.id', 'beat.verified_on' ]
+  };
+
+  const response = await callWithRequest('mget', params);
+  return get(response, 'docs', []);
+}
+
+async function verifyBeats(callWithRequest, beatIds) {
+  if (!Array.isArray(beatIds) || (beatIds.length === 0)) {
+    return [];
+  }
+
+  const verifiedOn = moment().toJSON();
+  const body = flatten(beatIds.map(beatId => [
+    { update: { _id: `beat:${beatId}` } },
+    { doc: { beat: { verified_on: verifiedOn } } }
+  ]));
+
+  const params = {
+    index: INDEX_NAMES.BEATS,
+    type: '_doc',
+    body,
+    refresh: 'wait_for'
+  };
+
+  const response = await callWithRequest('bulk', params);
+  return get(response, 'items', []);
+}
+
+function determineNonExistentBeatIds(beatsFromEs, beatIdsFromRequest) {
+  return beatsFromEs.reduce((nonExistentBeatIds, beatFromEs, idx) => {
+    if (!beatFromEs.found) {
+      nonExistentBeatIds.push(beatIdsFromRequest[idx]);
+    }
+    return nonExistentBeatIds;
+  }, []);
+}
+
+function determineAlreadyVerifiedBeatIds(beatsFromEs) {
+  return beatsFromEs
+    .filter(beat => beat.found)
+    .filter(beat => beat._source.beat.hasOwnProperty('verified_on'))
+    .map(beat => beat._source.beat.id);
+}
+
+function determineToBeVerifiedBeatIds(beatsFromEs) {
+  return beatsFromEs
+    .filter(beat => beat.found)
+    .filter(beat => !beat._source.beat.hasOwnProperty('verified_on'))
+    .map(beat => beat._source.beat.id);
+}
+
+function determineVerifiedBeatIds(verifications, toBeVerifiedBeatIds) {
```

Comment:
I'm wondering if we can simplify these function names.

Is it misleading to call them `find` instead of `determine`?

I've also tried a number of ways of simplifying the suffixes but I have come to the conclusion that they're pretty well-named already.

Reply:
I think `find` might be more readable.

I've approved the PR as everything else LGTM.